storage-manager/src/KVStorageInitializer.cpp

```diff
@@ -0,0 +1,74 @@
+/* Copyright (C) 2024 MariaDB Corporation
+
+   This program is free software; you can redistribute it and/or
+   modify it under the terms of the GNU General Public License
+   as published by the Free Software Foundation; version 2 of
+   the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+   MA 02110-1301, USA. */
+#include "Config.h"
+#include "SMLogging.h"
+#include "KVStorageInitializer.h"
+
+namespace storagemanager
+{
+static std::shared_ptr<FDBCS::FDBDataBase> fdbDataBaseInstance;
+static std::unique_ptr<FDBCS::FDBNetwork> fdbNetworkInstance;
+static std::mutex kvStorageLock;
+
+std::shared_ptr<FDBCS::FDBDataBase> KVStorageInitializer::getStorageInstance()
+{
+  if (fdbDataBaseInstance)
+    return fdbDataBaseInstance;
+
+  std::unique_lock<std::mutex> lock(kvStorageLock);
+  if (fdbDataBaseInstance)
+    return fdbDataBaseInstance;
+
+  auto* config = Config::get();
+  auto* logger = SMLogging::get();
+
+  if (!FDBCS::setAPIVersion())
+  {
```

Comment:
There are 4 times you use the same code here, mb. it is worth to put it into an inline func?